rustls/src/server/server_conn.rs

```diff
@@ -110,6 +110,16 @@ pub trait ProducesTickets: Debug + Send + Sync {
     fn decrypt(&self, cipher: &[u8]) -> Option<Vec<u8>>;
 }
 
+/// Selects TLS 1.3 external preshared keys.
+pub trait SelectsPresharedKeys: Debug + Send + Sync {
+    /// Retrieves a preshared key.
+    ///
+    /// To help prevent adversaries from discovering identities
+    /// supported by the server, identity comparisons should be
+    /// performed in constant time.
+    fn load_psk(&self, identity: &[u8]) -> Option<Arc<PresharedKey>>;
```

Comment:
If we need constant-time here, suggest we change the API so that rustls itself handles the identity comparison.